### PR TITLE
Add onnxblock large models support & allow loss block to accept extra inputs.

### PR DESCRIPTION
**Pull request by AdamLouly**
# onnxblock large models support.
This PR makes it possible for onnxblock to handle large models more effectively. 
Currently, onnxblock encounters problems when dealing with models larger than 2GB due to protobuf message max size. 
To address this issue, we temporarily remove initializers from the model during shape inference, and once shape inference is complete, the initializers are added back to the model. 

This change ensures that onnxblock can handle models larger than 2GB without issues.

# loss function extra inputs.

Currently, the loss functions in onnxblock expect exactly two inputs in their build method. 
Occasionally, models may pass additional inputs, causing the build function to fail. 
To solve this issue, we can temporarily resolve it by adding *args to the build method, allowing it to ignore any additional inputs beyond the expected two.

### CODE REVIEW

File: orttraining/orttraining/python/training/onnxblock/onnxblock.py [hunk 1]
```diff
@@ -187,7 +187,30 @@ def __call__(self, *args, **kwargs):
...
+        except ValueError as e:  # Catching ValueError exceptions
+            # Now, e is an instance of ValueError, and you can check its message
+            if "exceeds maximum protobuf size of 2GB" in str(e):
+                size_check = True
+
+        if size_check:
+            initializers_backup = model.graph.initializer[:]
+
+            # Remove initializers from the model
+            model.graph.ClearField("initializer")
```
Comment (tianleiwu) [inline orttraining/orttraining/python/training/onnxblock/onnxblock.py:205]: This is tricky. What if the shape inference need some shape related initializers?
I suggest to call another onnx API for model > 2GB:
```
onnx.shape_inference.infer_shapes_path(input_model_path, inferred_model_path)
```

File: orttraining/orttraining/python/training/onnxblock/onnxblock.py [hunk 2]
```diff
@@ -187,7 +187,30 @@ def __call__(self, *args, **kwargs):
 
         output = self.build(*args, **kwargs)
 
-        model = onnx.shape_inference.infer_shapes(accessor._GLOBAL_ACCESSOR.model)
+        model = accessor._GLOBAL_ACCESSOR.model
+
+        size_check = False
```
Comment (baijumeswani) [inline orttraining/orttraining/python/training/onnxblock/onnxblock.py:192]: Can we rename this variable to something like `model_exceeds_max_permissible_size`?
